packages/desktop/views/dashboard/delegation/Delegation.svelte

```diff
@@ -17,51 +17,49 @@
     import { formatTokenAmountBestMatch, AddressConverter, getClient, selectedWalletAssets } from '@core/wallet'
     import { truncateString } from '@core/utils'
     import { Icon as IconEnum } from '@auxiliary/icon'
-    import { OutputType, DelegationOutput, AccountAddress, OutputData } from '@iota/sdk/out/types'
+    import { OutputType, DelegationOutput, OutputData } from '@iota/sdk/out/types'
     import { PopupId, openPopup } from '@auxiliary/popup'
     import features from '@features/features'
 
     let delegationData: IDelegationTable[] = []
+    let currentEpoch = 0
 
     enum Header {
-        Name = 'name',
+        DelegationId = 'delegationId',
         DelegatedFunds = 'delegatedFunds',
         Rewards = 'rewards',
-        Epoch = 'epoch',
-        Address = 'address',
+        Epochs = 'epochs',
+        DelegatedAddress = 'delegatedAddress',
         Action = 'action',
     }
 
     interface IDelegationTable {
-        [Header.Name]: string
+        [Header.DelegationId]: string
         [Header.DelegatedFunds]: number
         [Header.Rewards]: number
-        [Header.Epoch]: number
-        [Header.Address]: string
+        [Header.Epochs]: number
+        [Header.DelegatedAddress]: string
         [Header.Action]: () => void
     }
 
-    // TODO: update interface when available
-    interface DelegationOutputTemp extends DelegationOutput {
-        validatorAddress: AccountAddress
-    }
-
     $: delegationOutputs =
         $selectedWallet?.walletUnspentOutputs?.filter((output) => output?.output?.type === OutputType.Delegation) || []
-    $: delegationOutputs?.length > 0 && mappedDelegationData(delegationOutputs)
+    $: delegationOutputs?.length > 0 && getCurrentEpoch()
+    $: delegationOutputs?.length > 0 && currentEpoch && mappedDelegationData(delegationOutputs)
     $: ({ baseCoin } = $selectedWalletAssets[$activeProfile?.network.id])
 
     async function mappedDelegationData(delegationOutputs: OutputData[]): Promise<void> {
         const result =
-            delegationOutputs?.map(async (output, index) => {
-                const delegationOutput = output.output as DelegationOutputTemp
+            delegationOutputs?.map(async (output) => {
+                const delegationOutput = output.output as DelegationOutput
+                // Until the first epoch in which it was delegated ends, no rewards are obtained
+                const epochsDelegating = currentEpoch - delegationOutput.startEpoch
                 return {
-                    [Header.Name]: `Delegation ${index + 1}`,
+                    [Header.DelegationId]: delegationOutput.delegationId,
                     [Header.DelegatedFunds]: Number(delegationOutput.delegatedAmount),
                     [Header.Rewards]: await getOutputRewards(output.outputId),
```

Comment:
`getOutputRewards` should not live in this component, i would move that logic out so its reusable somewhere else, these types of logics dont make sense in a svelte component